File: example3.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program
{
    static void Main(string[] args)
    {
        string str = "aabbccdefgh";
        Dictionary<char, int> frequency = new Dictionary<char, int>();

        foreach (char c in str)
        {
            if (!frequency.ContainsKey(c))
            {
                frequency[c] = 1;
            }
            else
            {
                frequency[c]++;
            }
        }

        var nonRepeatingChars = frequency.Where(pair => pair.Value == 1).Select(pair => pair.Key);

        Console.WriteLine("Non-repeating characters in '" + str + "': " + string.Join(", ", nonRepeatingChars));
    }
}

File: example3.java
import java.util.HashMap;
import java.util.Map;

public class Main {
    public static void main(String[] args) {
        String str = "aabbccdefgh";
        Map<Character, Integer> frequency = new HashMap<>();

        for (char c : str.toCharArray()) {
            frequency.put(c, frequency.getOrDefault(c, 0) + 1);
        }

        StringBuilder nonRepeatingChars = new StringBuilder();
        for (Map.Entry<Character, Integer> entry : frequency.entrySet()) {
            if (entry.getValue() == 1) {
                nonRepeatingChars.append(entry.getKey()).append(", ");
            }
        }

        if (nonRepeatingChars.length() > 0) {
            nonRepeatingChars.delete(nonRepeatingChars.length() - 2, nonRepeatingChars.length());
        }

        System.out.println("Non-repeating characters in '" + str + "': " + nonRepeatingChars.toString());
    }
}
